scripts/subset_matrix_table.py

```diff
@@ -140,28 +140,41 @@ def clean_locus(contig: str, pos: str) -> hl.IntervalExpression | None:
 
     Returns
     -------
-    A parsed hail locus. For a point change this will be the result of
-    parsing "contig:pos-pos+1"
+    A parsed hail locus. For a point change this will be contig:pos-pos+1
     """
-    if not all([contig, pos]):
+    if not any([contig, pos]):
         return None
 
-    if '-' in pos:
-        start, end = map(int, pos.split('-'))
+    if pos and not contig:
+        raise Exception(f'Positional filtering requires a chromosome')
+
+    if contig and not pos:
+        start = 'start'
+        end = 'end'
+
+    elif '-' in pos:
+        assert (
+            pos.count('-') == 1
+        ), f'Positions must be one value, or a range between two values: {pos}'
+        start, end = pos.split('-')
+        if start != 'start':
+            assert int(start), f'start value could not be converted to an int: {start}'
+            if int(start) < 1:
+                start = 1
+        if end != 'end':
+            assert int(end), f'end value could not be converted to an int: {end}'
+            # adjust the end value if it is out of bounds
+            if int(end) > hl.get_reference('GRCh38').lengths[contig]:
+                end = hl.get_reference('GRCh38').lengths[contig]
 
-        # quick validation that we only received 2 values
-        assert isinstance(start, int)
-        assert isinstance(end, int)
-
-        assert start <= end
-
-        if start == end:
-            end += 1
     else:
+        assert int(
+            pos
+        ), f'if only one position is specified, it must be numerical: {pos}'
         start = int(pos)
         end = start + 1
 
-    return hl.parse_locus_interval(f'{contig}:{start}-{end}')
+    return hl.parse_locus_interval(f'{contig}:{start}-{end}', reference_genome='GRCh38')
```

Comment:
You've dropped the `start < end` check if they're both ints which would be useful IMO to keep.

Reply:
Updated to re-add this, thanks!